### PR TITLE
OpenGL 3.3 Support

### PR DESCRIPTION
**Pull request by Berulacks**
Title says it all. We use regex to remove layout qualifiers from our shader strings before compilation according to the Graphics/"OpenGL3.3" flag in the configuration file.

Also disabled rendering the ui on Linux until we get that fixed.

### CODE REVIEW

File: source/dash/graphics/adapters/sdl.d [hunk 1]
```diff
@@ -47,6 +47,18 @@ public:
 
         window.setTitle( DGame.instance.title );
 
+    if(config.graphics.usingGl33)
```
Comment (dislogical) [inline source/dash/graphics/adapters/sdl.d]: The spacing here looks a little suspect. Double check your settings?

File: source/dash/graphics/shaders/shaders.d [hunk 2]
```diff
@@ -140,6 +140,8 @@ final package class Shader
 private:
     uint _programID, _vertexShaderID, _fragmentShaderID;
     string _shaderName;
+    auto versionRegex = ctRegex!r"\#version\s400";
+    auto layoutRegex = ctRegex!r"layout\(location\s\=\s[0-9]\)\s";
```
Comment (dislogical) [inline source/dash/graphics/shaders/shaders.d]: This may cause problems if we get higher than 9 `in`s. Can we make it `[0-9]+`?

**Comment by Berulacks** [comment]
Sure. 1 sec.

File: source/dash/utility/config.d [hunk 3]
```diff
@@ -67,6 +67,8 @@ public:
         bool backfaceCulling = true;
         @rename( "VSync" ) @optional
         bool vsync = false;
+        @rename( "OpenGL3.3" ) @optional
+	    bool usingGl33 = false;
```
Comment (dislogical) [inline source/dash/utility/config.d]: Check spacing here. Also, maybe do the rename to `OpenGL33`, I'm not sure how much the various file formats we use like `.`s.

**Comment by dislogical** [comment]
Aside from the nits, looks good.

**Comment by Berulacks** [comment]
How are the nits looking, now?

**Comment by dislogical** [comment]
I'll merge when green.

**Pull request merged**